### PR TITLE
sync az and azd for terraform pipeline definition

### PR DESCRIPTION
**Pull request by vhvb1989**
fix: https://github.com/Azure/azure-dev/issues/2231

Adding `az` login and account setup for terraform's pipeline definitions.
This is required after we added `az` commands to the end of `azd provision` for terraform templates to set up basic auth.

This was not catch during CI because we did add the az and az account sync to test-templates CI.  However, we also need this fix for the GitHub pipeline definition.

We don't need to do this for the Azure DevOps pipeline definition because it still uses az-cli auth and automatically sets the az-account based on the connection.

### CODE REVIEW

**Review by wbreza — APPROVED** [review]

File: templates/common/.github/workflows/java/terraform/azure-dev.yml [hunk 1]
```diff
@@ -28,6 +28,24 @@ jobs:
...
+      - name: Login az
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Set az account
+        uses: azure/CLI@v1
+        with:
+          inlineScript: |
+            az account show
```
Comment (weikanglim) [inline templates/common/.github/workflows/java/terraform/azure-dev.yml:45]: Do we need to run `account show` before and after?

File: templates/common/.github/workflows/java/terraform/azure-dev.yml [hunk 2]
```diff
@@ -28,6 +28,24 @@ jobs:
       - run: |
           chmod +x ${GITHUB_WORKSPACE}/src/api/mvnw
 
+      - name: Install Nodejs
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
```
Comment (weikanglim) [inline templates/common/.github/workflows/java/terraform/azure-dev.yml:34]: Why install node for java?

Comment (vhvb1989) [reply templates/common/.github/workflows/java/terraform/azure-dev.yml:34]: We always need `node` to build the front-end application, regardless of the api-language

File: templates/common/.github/workflows/java/terraform/azure-dev.yml (hunk 1 above)
Comment (vhvb1989) [reply templates/common/.github/workflows/java/terraform/azure-dev.yml:45]: Will update. 
That was part of me testing it :)

File: templates/common/.github/workflows/java/terraform/azure-dev.yml (hunk 2 above)
Comment (weikanglim) [reply templates/common/.github/workflows/java/terraform/azure-dev.yml:34]: ah

**Pull request merged**